cmake/kokkos_enable_options.cmake

```diff
@@ -130,6 +130,22 @@ IF (KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION)
   SET(KOKKOS_OPT_RANGE_AGGRESSIVE_VECTORIZATION ON)
 ENDIF()
 
+# Force consistency of KOKKOS_ENABLE_CUDA_RELOCATABLE_DEVICE_CODE
+# and CMAKE_CUDA_SEPARABLE_COMPILATION when we are compiling
+# using the CMake CUDA language support.
+# Either one being on will turn the other one on.
+IF (KOKKOS_COMPILE_LANGUAGE STREQUAL CUDA)
+  IF (KOKKOS_ENABLE_CUDA_RELOCATABLE_DEVICE_CODE)
+    IF (NOT CMAKE_CUDA_SEPARABLE_COMPILATION)
+      SET(CMAKE_CUDA_SEPARABLE_COMPILATION ON)
+    ENDIF()
```

Comment:
I think we might want to raise a warning (or display a STATUS message) when setting the `CMAKE_CUDA_SEPARABLE_COMPILATION` variable